controllers/machinepool.go

```diff
@@ -45,7 +49,7 @@ func (r *GarbageCollectorController) Reconcile(ctx context.Context, req ctrl.Req
 	if err != nil {
 		if k8sErrors.IsNotFound(err) {
 			logger.Info("MachinePool no longer exists")
-			return ctrl.Result{}, nil
+			return ctrl.Result{Requeue: false}, nil
```

Comment:
This should default to `false` anyway, no?

Reply:
Yeah you're right. Fixed